Assets/HoloToolkit/Speech/Scripts/Wav.cs

```diff
@@ -0,0 +1,152 @@
+        /// Initializes a new <see cref="Wav"/> instance.
+        /// </summary>
+        /// <param name="wav">
+        /// The raw WAV byte data.
+        /// </param>
+        public Wav(byte[] wav)
+        {
+            // Determine if mono or stereo
+            ChannelCount = wav[22];     // Forget byte 23 as 99.999% of WAVs are 1 or 2 channels
+
+            // Get the frequency
+            Frequency = bytesToInt(wav, 24);
+
+            // Get past all the other sub chunks to get to the data subchunk:
+            int pos = 12;   // First subchunk ID from 12 to 16
+
+            // Keep iterating until we find the data chunk (i.e. 64 61 74 61 ...... (i.e. 100 97 116 97 in decimal))
+            while (!(wav[pos] == 100 && wav[pos + 1] == 97 && wav[pos + 2] == 116 && wav[pos + 3] == 97))
+            {
+                pos += 4;
+                int chunkSize = wav[pos] + wav[pos + 1] * 256 + wav[pos + 2] * 65536 + wav[pos + 3] * 16777216;
+                pos += 4 + chunkSize;
+            }
+            pos += 8;
+
+            // Pos is now positioned to start of actual sound data.
+            SampleCount = (wav.Length - pos) / 2;     // 2 bytes per sample (16 bit sound mono)
+            if (ChannelCount == 2) SampleCount /= 2;        // 4 bytes per sample (16 bit stereo)
+
+            // Allocate memory (right will be null if only mono sound)
+            LeftChannel = new float[SampleCount];
+            if (ChannelCount == 2) RightChannel = new float[SampleCount];
+            else RightChannel = null;
+
+            // Write to double array/s:
+            int i = 0;
+            while (pos < wav.Length)
+            {
+                LeftChannel[i] = bytesToFloat(wav[pos], wav[pos + 1]);
+                pos += 2;
+                if (ChannelCount == 2)
+                {
+                    RightChannel[i] = bytesToFloat(wav[pos], wav[pos + 1]);
+                    pos += 2;
+                }
+                i++;
+            }
+        }
+        #endregion // Constructors
+
+        #region Overrides / Event Handlers
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return string.Format("[WAV: LeftChannel={0}, RightChannel={1}, ChannelCount={2}, SampleCount={3}, Frequency={4}]", LeftChannel, RightChannel, ChannelCount, SampleCount, Frequency);
+        }
+        #endregion // Overrides / Event Handlers
+
+        #region Public Methods
+        /// <summary>
+        /// Dynamically creates an <see cref="AudioClip"/> that represents the WAV file.
+        /// </summary>
+        /// <param name="name">
+        /// The name of the dynamically generated clip.
+        /// </param>
+        /// <returns>
+        /// The <see cref="AudioClip"/>.
+        /// </returns>
+        public AudioClip ToClip(string name)
+        {
+            // Create the audio clip
+            var clip = AudioClip.Create(name, SampleCount, 1, Frequency, false); // TODO: Support stereo
```

Comment:
Unity requires the audio to be interleaved (I believe every 4 samples if I remember correctly). You would set the 3rd parameter in the Create method above to 2 and you would interleave the audio samples L R L R L R etc.

For TextToSpeech and for spatial audio there's not much of a reason to do this because TTS is usually mono and Unity cannot do spatial placement with stereo audio clips. I have the TODO in there in case we wanted to make the Wav class a general purpose class. But that may very well be overkill for the the toolkit.